### ISSUE TITLE
Regression: unable to specify a managed disk to use as os_disk with the new azurerm_windows_virtual_machine or azurerm_linux_virtual_machine

### ISSUE THREAD

**Issue by ntherrien** [terraform-providers/terraform-provider-azurerm#7269]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.12.26
+ provider.azurerm v2.13.0

### Affected Resource(s)

azurerm_linux_virtual_machine and azurerm_windows_virtual_machine

### Expected Behavior

This is a regression from azurerm_virtual_machine. I am reported this as a bug because the documentation says that azurerm_virtual_machine has been superseded by the newer linux and windows versions and that the azurerm_virtual_machine will be removed in future versions of terraform...  

The previous azurerm_virtual_machine supports creating a vm from a specialized managed disk:

```
resource "azurerm_managed_disk" "aOsDisk" {
  name                 = "aOsDisk"
  location             = var.location
  resource_group_name  = var.resource_group_name
  storage_account_type = "Standard_LRS"
  create_option        = "Import"
  os_type              = "Linux"
  source_uri = "https://URI.blob.core.windows.net/someimage.vhd"
  disk_size_gb         = "256"

}
resource "azurerm_virtual_machine" "a" {
  name =  "a"
  location = var.location
  resource_group_name   = var.resource_group_name
  network_interface_ids = [azurerm_network_interface.a-nic.id]
  vm_size = "Standard_F4s_v2"

  storage_os_disk {
    name            = "aOsDisk"
    managed_disk_id = azurerm_managed_disk.aOsDisk.id
    caching           = "ReadWrite"
    create_option     = "Attach"
    os_type = "Linux"
    disk_size_gb = 256
  }

}
```

### Actual Behavior

No documented option to attach an osdisk.

### References

https://www.terraform.io/docs/providers/azurerm/r/linux_virtual_machine.html
https://www.terraform.io/docs/providers/azurerm/r/virtual_machine.html

**Comment by ArcturusZhang** [comment]
Hi @ntherrien thanks for opening this issue!

By the 4th `Note` in the [document](https://www.terraform.io/docs/providers/azurerm/r/linux_virtual_machine.html) of `azurerm_linux_virtual_machine`, this resource is by-design not support attachment of the os disks. You could instead use an image of the os disk, or continuing use the old `azurerm_virtual_machine` resource.

**Comment by tombuildsstuff** [comment]
hey @ntherrien 

As @ArcturusZhang has mentioned whilst the older `azurerm_virtual_machine` resource has been around for a while and has accumulated support for the different generations of disks (e.g. unmanaged vs managed), disk encryption (e.g. disk encryption sets vs using the VM Extension) - the new Virtual Machine resources instead only support the latest generation of each approach.

When these resources were built, the latest recommendations were to use a Shared Image Gallery where possible, although we also support provisioning from an existing image outside of a Shared Image Gallery using the `source_image_id` parameter - instead of creating a disk ahead of time.

Whilst at this time we don't support attaching an existing OS Disk to a Virtual Machine with these resources, this is something we could look to support in these resources in the future - if you're looking for support for this would you mind opening a Feature Request to track this?

It's worth noting that if you're looking to keep the disks around, it's possible to keep the OS Disk around using [the `delete_os_disk_on_deletion` flag within the `features` block](https://www.terraform.io/docs/providers/azurerm/index.html#features-2) to configure this behaviour.

Since this behaviour is currently by design, I'm going to close this issue for the moment - however as mentioned above if you're looking for support for attaching an existing OS Disk to the new Virtual Machine resources then please open a Feature Request and we can take another look 👍 

Thanks!

**Comment by ntherrien** [comment]
Thanks @tombuildsstuff and @ArcturusZhang  for your replies.   I am okay with creating a Feature Request but my concern is if the azurerm_virtual_machine is becoming deprecated and that the linux and windows variants are getting new features that the old method doesn't have, there will be a growing discrepancy between them.  This is why I tried to draw attention on this by raising it as a regression bug instead of a feature request. If it is supported on the older resource type, it should be supported on the new resource types (or the older resource type needs to stay and be upgraded with the same features available on the new resource types).

Specialized OS Disks are the only way to introduce legacy ISO based installs to the cloud.  We cannot boot from a cd-rom in Azure so we must do this locally and then upload the VHD. That's how base (generalized) images are created, too.   Sometimes its just not possible to generalize the images, in particular when trying to lift an on-premise system to the cloud, which is what lead me to notice this discrepancy in the first place (because I tried using linux_vm resource with specialized os disk).

I understand that the more and more time we work with cloud resources, the more and more tempting it is to get rid of older stuff, simply because better options are now available. But I would appreciate it if we could maintain support for Specialized images in future releases by making sure the new resource types fully support them. 

As for the Feature Request you asked me to create, should I simply copy this information into a new request, or should we change the type of this current request and reopen?

Cheers,
